mmv1/third_party/terraform/services/bigtable/resource_bigtable_table.go

```diff
@@ -103,6 +103,29 @@ func ResourceBigtableTable() *schema.Resource {
 				ValidateFunc: verify.ValidateDuration(),
 				Description:  `Duration to retain change stream data for the table. Set to 0 to disable. Must be between 1 and 7 days.`,
 			},
+
+			"automated_backup_policy": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"retention_period": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: verify.ValidateDuration(),
+							Description:  `How long the automated backups should be retained. The only supported value at this time is 3 days.`,
+						},
+						"frequency": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: verify.ValidateDuration(),
```

Comment:
how does the validation work?
For garbage collection policy, we created a new~ ParseDuration function that can handled weeks (I think only weeks? maybe more). Do we want to use that one instead of the time.ParseDuration? If so, I'm not sure what ValidateDuration does, but maybe we want to do the same for the validation (it'd be weird if it didn't pass validation, but was valid everywhere else...)

Reply:
Thanks

I'm reusing the same `verify.ValidateDuration` that `change_stream_retention` above uses (which also parses using `time.ParseDuration`) https://github.com/ma-g-22/magic-modules/blob/ab/mmv1/third_party/terraform/services/bigtable/resource_bigtable_table.go#L103

Is this the new `ParseDuration` function? https://github.com/ma-g-22/magic-modules/blob/ab/mmv1/third_party/terraform/services/bigtable/duration_parsing_helper.go#L26 - I'm not sure if we need weeks specifically?

Reply:
Thanks, updated to use `bigtable.ParseDuration`